packages/hash/api/src/model/knowledge/comment.model.ts

```diff
@@ -0,0 +1,224 @@
+import { WORKSPACE_TYPES } from "../../graph/workspace-types";
+
+type CommentModelCreateParams = Omit<
+  EntityModelCreateParams,
+  "properties" | "entityTypeModel"
+> & {
+  createdBy: UserModel;
+  parent: EntityModel;
+  tokens: TextToken[];
+};
+
+/**
+ * @class {@link CommentModel}
+ */
+export default class extends EntityModel {
+  static fromEntityModel(entity: EntityModel): CommentModel {
+    if (
+      entity.entityTypeModel.schema.$id !==
+      WORKSPACE_TYPES.entityType.comment.schema.$id
+    ) {
+      throw new Error(
+        `Entity with id ${entity.entityId} is not a workspace comment`,
+      );
+    }
+
+    return new CommentModel(entity);
+  }
+
+  /**
+   * Get a workspace comment entity by its entity id.
+   *
+   * @param params.entityId - the entity id of the comment
+   */
+  static async getCommentById(
+    graphApi: GraphApi,
+    params: { entityId: string },
+  ): Promise<CommentModel> {
+    const entity = await EntityModel.getLatest(graphApi, params);
+
+    return CommentModel.fromEntityModel(entity);
+  }
+
+  /**
+   * Create a workspace block entity.
+   *
+   * @param params.componentId - the component id of the block
+   * @param params.blockData - the linked block data entity
+   * @see {@link EntityModel.create} for remaining params
+   */
+  static async createComment(
+    graphApi: GraphApi,
+    params: CommentModelCreateParams,
+  ): Promise<CommentModel> {
+    const { ownedById } = params;
+
+    const entityTypeModel = WORKSPACE_TYPES.entityType.comment;
+
+    const entity = await EntityModel.create(graphApi, {
+      ownedById,
+      properties: {},
+      entityTypeModel,
+    });
+
+    const { tokens, parent, createdBy } = params;
+
+    const textEntity = await EntityModel.create(graphApi, {
+      ownedById,
+      properties: {
+        [WORKSPACE_TYPES.propertyType.tokens.baseUri]: tokens,
+      },
+      entityTypeModel: WORKSPACE_TYPES.entityType.text,
+    });
+
+    await entity.createOutgoingLink(graphApi, {
+      linkTypeModel: WORKSPACE_TYPES.linkType.hasText,
+      targetEntityModel: textEntity,
+      ownedById,
+    });
+
+    await entity.createOutgoingLink(graphApi, {
+      linkTypeModel: WORKSPACE_TYPES.linkType.parent,
+      targetEntityModel: parent,
+      ownedById,
+    });
+
+    await entity.createOutgoingLink(graphApi, {
+      linkTypeModel: WORKSPACE_TYPES.linkType.author,
+      targetEntityModel: createdBy,
+      ownedById,
+    });
+
+    return CommentModel.fromEntityModel(entity);
+  }
+
+  static async getAllCommentsInPage(
+    graphApi: GraphApi,
+    params: {
+      ownedById: string;
+      pageId: string;
+    },
+  ): Promise<CommentModel[]> {
+    const { ownedById, pageId } = params;
+
+    const pageEntity = await PageModel.getPageById(graphApi, {
+      entityId: pageId,
+    });
+
+    if (!pageEntity) {
+      throw new Error(
+        `Page with entityId ${pageId} not found found in account ${ownedById}`,
+      );
+    }
```

Comment:
Dead code
```suggestion
    const pageEntity = await PageModel.getPageById(graphApi, {
      entityId: pageId,
    });
```

Reply:
This was removed as part of 209e4542a62621996b31731bc43617081c8ed35e

Reply:
Perhaps this hasn't been pushed yet, but as far as I can tell from the Github UI this has not been addressed.

Reply:
Sorry it slipped through. Actually removed it in 26cf592450772af0b3517b3136f7e0da641a1a4e